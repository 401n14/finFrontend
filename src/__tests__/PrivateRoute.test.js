import React from 'react';
import PrivateRoute from '../components/PrivateRoute';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import { useAuth0 } from '../react-auth0-spa.js';

configure({adapter:new Adapter()});

jest.mock('../react-auth0-spa.js');

function testComp(){
    return (
        <div>
            Test
        </div>
    )
}
describe('Loading Tests', () => {
    const wrapper = shallow(
<PrivateRoute />
    );
    it('Exists on Render', () => {
        expect(wrapper).toBeTruthy();
    });
});
describe('Snapshot Test', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            loginWithRedirect: jest.fn(),
        });
    });
    it('renders correctly logged in, Snapshot', () => {
        const page = renderer.create(
        <BrowserRouter>
        <PrivateRoute 
        path="/" exact component={testComp}
        />
        </BrowserRouter>
        ).toJSON();
        expect(page).toMatchSnapshot();
    });
});
