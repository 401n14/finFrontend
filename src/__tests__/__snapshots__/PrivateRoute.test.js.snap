// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly logged in, Snapshot 1`] = `
<div>
  Test
</div>
`;
